.github/workflows/preview.yml

```diff
@@ -0,0 +1,85 @@
+name: preview
+
+on:
+  workflow_call: ~
+
+permissions:
+  id-token: write
+  pull-requests: write
+  deployments: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Deployment
```

Comment:
Not necessary for this PR but we might want to protect this to only run if doc changes are part of the PR?

Reply:
My intention for this preview was not to create a preview of the docs only, but actually also a preview of changes within the docs-builder codebase.